dawn/src/dawn/CodeGen/CXXNaive/ASTStencilBody.cpp

```diff
@@ -32,23 +33,21 @@ ASTStencilBody::~ASTStencilBody() {}
 
 std::string ASTStencilBody::getName(const std::shared_ptr<iir::Stmt>& stmt) const {
   if(currentFunction_)
-    return currentFunction_->getFieldNameFromAccessID(currentFunction_->getAccessIDFromStmt(stmt));
+    return currentFunction_->getFieldNameFromAccessID(
+        *stmt->getData<iir::VarDeclStmtData>().AccessID);
   else
-    return metadata_.getFieldNameFromAccessID(metadata_.getAccessIDFromStmt(stmt));
+    return metadata_.getFieldNameFromAccessID(*stmt->getData<iir::VarDeclStmtData>().AccessID);
 }
 
 std::string ASTStencilBody::getName(const std::shared_ptr<iir::Expr>& expr) const {
   if(currentFunction_)
-    return currentFunction_->getFieldNameFromAccessID(currentFunction_->getAccessIDFromExpr(expr));
+    return currentFunction_->getFieldNameFromAccessID(iir::getAccessIDFromExpr(expr));
   else
-    return metadata_.getFieldNameFromAccessID(metadata_.getAccessIDFromExpr(expr));
+    return metadata_.getFieldNameFromAccessID(iir::getAccessIDFromExpr(expr));
 }
 
 int ASTStencilBody::getAccessID(const std::shared_ptr<iir::Expr>& expr) const {
-  if(currentFunction_)
-    return currentFunction_->getAccessIDFromExpr(expr);
-  else
-    return metadata_.getAccessIDFromExpr(expr);
+  return iir::getAccessIDFromExpr(expr);
```

Comment:
What about removing `ASTStencilBody::getAccessID()` instead? Do we really need the interface? It made sense before, because the information could have come from different contexts, but now it is always inside the stmt/expr and we can use the global helper functions.

Reply:
yes I think you're right. When you add the global function, you could probably also rename it to `iir::getAccessID`, because `FromStmt` and `FromExpr` does not add any information (as it is only a hint for the argument type that you pass anyway)

Reply:
ok